File: android/app/src/main/swift/Sources/WeatherCoreBridge/SwiftContainer.swift
//
// Created by Andriy Druk on 13.09.2020.
//

import Foundation
import WeatherCore

public class SwiftContainer {

    private let container: WeatherCoreContainer

    public init(basePath: String, apiKey: String) {
        self.container = WeatherCoreContainer(basePath: basePath, apiKey: apiKey)
    }

    public func getWeatherViewModel(delegate: LocationWeatherViewModelDelegate) -> LocationWeatherViewModel {
        return container.getWeatherViewModel(delegate: delegate)
    }

    public func getLocationSearchViewModel(delegate: LocationSearchDelegate) -> LocationSearchViewModel {
        return container.getLocationSearchViewModel(delegate: delegate)
    }

}

File: android/app/src/main/java/com/readdle/weather/core/SwiftContainer.kt
package com.readdle.weather.core

import com.readdle.codegen.anotation.SwiftFunc
import com.readdle.codegen.anotation.SwiftReference
import java.lang.annotation.Native

@SwiftReference
class SwiftContainer private constructor() {

    // Swift JNI private native pointer
    @Native
    private val nativePointer = 0L

    // Swift JNI release method
    external fun release()

    @SwiftFunc("getWeatherViewModel(delegate:)")
	external fun getWeatherViewModel(delegate: LocationWeatherViewModelDelegateAndroid): LocationWeatherViewModel

	@SwiftFunc("getLocationSearchViewModel(delegate:)")
	external fun getLocationSearchViewModel(delegate: LocationSearchDelegateAndroid): LocationSearchViewModel

    companion object {
        @JvmStatic @SwiftFunc("init(basePath:apiKey:)")
		external fun init(basePath: String, apiKey: String): SwiftContainer
    }

}